dungeon/src/contrib/hud/UIUtils.java

```diff
@@ -64,30 +71,89 @@ public static void center(final Actor actor) {
+   *
+   * <p>This function can be used to soft-wrap texts to a maximum of {@link UIUtils#MAX_ROW_LENGTH}
+   * characters per line. The text is sanitised before wrapping by replacing multiple consecutive
+   * whitespace characters (including line breaks) with single spaces. The text is then wrapped
+   * according to the following algorithm: The last space before the maximum line length of {@link
+   * UIUtils#MAX_ROW_LENGTH} characters is replaced by a line break. Any words exceeding the maximum
+   * line length are cut off and wrapped at the position of the maximum line length.
+   *
+   * @param text text to be soft-wrapped at {@link UIUtils#MAX_ROW_LENGTH} characters per line
+   * @return the reformatted text where all lines have been soft-wrapped to at maximum {@link
+   *     UIUtils#MAX_ROW_LENGTH} characters per line
+   */
+  public static String formatString(final String text) {
+    return formatString(text, MAX_ROW_LENGTH);
+  }
+
+  /**
+   * Wrap texts to a maximum line length of {@code maxLen} characters.
+   *
+   * <p>This function can be used to soft-wrap texts to a maximum of {@code maxLen} characters per
+   * line. The text is sanitised before wrapping by replacing multiple consecutive whitespace
+   * characters (including line breaks) with single spaces. The text is then wrapped according to
+   * the following algorithm: The last space before the maximum line length of {@code maxLen}
+   * characters is replaced by a line break. Any words exceeding the maximum line length are cut off
+   * and wrapped at the position of the maximum line length.
    *
-   * @param string String which should be reformatted.
+   * @param text text to be soft-wrapped at {@code maxLen} characters per line
+   * @param maxLen maximum number of characters per line
+   * @return the reformatted text where all lines have been soft-wrapped to at maximum {@code
+   *     maxLen} characters per line
    */
-  public static String formatString(final String string) {
-    StringBuilder formattedMsg = new StringBuilder();
-    String[] lines = string.split(System.lineSeparator());
-
-    for (String line : lines) {
-      String[] words = line.split(" ");
-      int sumLength = 0;
-
-      for (String word : words) {
-        sumLength += word.length();
-        formattedMsg.append(word);
-        formattedMsg.append(" ");
-
-        if (sumLength > MAX_ROW_LENGTH) {
-          formattedMsg.append(System.lineSeparator());
-          sumLength = 0;
-        }
+  public static String formatString(final String text, int maxLen) {
+    if (text == null) {
+      return null;
+    }
+
+    final boolean softWordWrap = true;
+
+    final StringBuilder sb = new StringBuilder();
+    final String[] words = text.trim().replaceAll("\\s+", " ").split(" ");
+
+    int wordIndex = 0;
+    int lineIndex = 0;
+
+    while (wordIndex < words.length) {
+      final String word = words[wordIndex++];
+      final int before = sb.length();
```

Comment:
```suggestion
      int before = sb.length();
```

Reply:
`final` auf primitive datentypen?

Reply:
du meinst bei lokalen variablen... das kann man machen, muss man aber nicht.

Reply:
nee, schon "primitive datentypen". afaik macht `final` bei klassen sinn, aber bei `int`... nicht wirklich.

Reply:
> nee, schon "primitive datentypen". afaik macht `final` bei klassen sinn, aber bei `int`... nicht wirklich.
> 
> edit: aber wenn du wirklich wert auf die unveränderlichkeit legst an der stelle, dann los ...

ich sehe das anders. gerade bei primitiven datentypen (+ string) macht es in java sinn, final zu verwenden, wenn der konkrete wert wirklich unveränderlich sein soll.

auf der anderen seite ist es aber eine frage des codestyles des jeweiligen projekts, und es gibt keine feste regel dazu, ob man lokale variablen als final definieren sollte, um damit auszudrücken oder hervorzuheben, dass sich ein bestimmter wert innerhalb eines blocks nicht ändert, bzw. nicht ändern darf, und um damit auch den ablauf einer schleife besser verständlich zu machen. manche halten das für "zu viel" und manche verwenden das "gerne". es bleibt wohl dem autor dann überlassen, das final "bei bedarf" zu verwenden.

jedenfalls, um mal einen punkt zu machen, ich kann deine vorschläge übernehmen und es an den zwei stellen wieder entfernen.